src/cli.rs

```diff
@@ -793,10 +807,10 @@ impl Opts {
                 anyhow::Result::<_, anyhow::Error>::Ok((name, diff))
             });
 
-        stream::iter(update_iter)
+        let differences = stream::iter(update_iter)
             .buffer_unordered(opts.max_concurrent_downloads)
-            .try_for_each(|(name, diff)| {
-                fn write_diff(writer: &mut impl Write, name: &str, diff: Vec<diff::DiffEntry>) {
+            .try_filter(|(name, diff)| {
```

Comment:
Since we don't use them afterwards now you could just use [try_any](https://docs.rs/futures/latest/futures/stream/trait.TryStreamExt.html#method.try_any) and rename the var to `has_diff` like it was before.

Reply:
`try_any` is neat. I had looked for `try_count` which doesn't exist but since I always want to check non-empty this makes more sense

Reply:
Oh wait no. `try_any` short-circuits which we absolutely do not want here

Reply:
argh! Maybe just return to just having a mutable bool? I think it's a bit weird to collect a vec just to not use the values inside them

Reply:
Though *could* also use `try_all` since you can invert the logic here to instead have like a `no_diffs`.

Reply:
I've thought of that, but `try_all` also short-circuits. After a brief detour over `try_fold` which turned out to be like a `mut has_diff` but strictly worse, I'm indeed now back to the mutable bool soltuion. Also has the shortest diff anyways